userland/shell/sh.c

```
@@ -307,11 +311,17 @@ int main(int argc, char *argv[]) {
         exit(0);
     }
 
-    /* VM mode: confine all host file I/O to the launch directory (or temp VM sandbox) */
+    /* VM mode: confine host file I/O to vm_hostfs/ under the launch directory (or temp VM sandbox). */
     if (vm_configure_root_from_cwd() != 0)
         fprintf(stderr, "[VM] 5-layer driver config warning: layer 4 shell/VM root is not configured\n");
     fs_jail_init();
 
+    /* Default host volume: ensure drive.img exists before block driver probes it. */
+    if (strcmp(current_disk_file, "drive.img") == 0 && access(current_disk_file, F_OK) != 0) {
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)